librz/core/cmd/cmd_print.c

```diff
@@ -2206,782 +2156,149 @@ RZ_IPI RzCmdStatus rz_cmd_print_magic_handler(RzCore *core, int argc, const char
-		}
-		rz_core_analysis_stats_free(as);
-		return ptr;
-	}
-	for (i = 0; i < nblocks; i++) {
-		if (rz_cons_is_breaked()) {
-			break;
+	if (origin != offset) {
+		if (mode == RZ_OUTPUT_MODE_RIZIN) {
+			rz_cons_printf("CC-@ 0x%08" PFMT64x "\n", origin);
+			rz_cons_printf("CC switch table @ 0x%08" PFMT64x "\n", origin);
+			rz_cons_printf("axd 0x%" PFMT64x " @ 0x%08" PFMT64x "\n", origin, offset);
 		}
-		ut64 off = from + (i + skipblocks) * blocksize;
-		for (j = 0; j < blocksize; j++) {
-			if (type == 'a') {
-				RzAnalysisFunction *fcn = rz_analysis_get_fcn_in(core->analysis, off + j, 0);
-				if (fcn) {
-					ptr[i] = rz_list_length(fcn->bbs);
-				}
-				continue;
-			}
-			RzAnalysisOp *op = rz_core_analysis_op(core, off + j, RZ_ANALYSIS_OP_MASK_BASIC);
-			if (op) {
-				if (op->size < 1) {
-					// do nothing
-					if (type == 'i') {
-						matchBar(ptr, i);
-					}
-				} else {
-					if (checkAnalType(op, type)) {
-						matchBar(ptr, i);
-					}
-				}
-				if (op->size > 0) {
-					j += op->size - 1;
-				}
-				rz_analysis_op_free(op);
-			} else {
-				if (type == 'i') {
-					matchBar(ptr, i);
-				}
+	}
+	int n = 0;
+	for (int i = 0; (i + sizeof(st32)) <= len; i += step, n++) {
+		st32 *delta = (st32 *)(buf + i);
+		ut64 addr = offset + *delta;
+		if (!rz_io_is_valid_offset(core->io, addr, 0)) {
+			// Lets check for jmptbl with not relative addresses
+			// Like: jmp dword [eax*4 + jmptbl.0x5435345]
+			if (!rz_io_is_valid_offset(core->io, *delta, 0)) {
+				break;
 			}
+			addr = *delta;
+		}
+		if (mode == RZ_OUTPUT_MODE_RIZIN) {
+			rz_cons_printf("af case.%d.0x%" PFMT64x " 0x%08" PFMT64x "\n", n, offset, addr);
+			rz_cons_printf("ax 0x%" PFMT64x " @ 0x%08" PFMT64x "\n", offset, addr);
+			rz_cons_printf("ax 0x%" PFMT64x " @ 0x%08" PFMT64x "\n", addr, offset); // wrong, but useful because forward xrefs dont work :?
+			// FIXME: "aho" doesn't accept anything here after the "case" word
+			rz_cons_printf("aho case 0x%" PFMT64x " 0x%08" PFMT64x " @ 0x%08" PFMT64x "\n", (ut64)i, addr, offset + i); // wrong, but useful because forward xrefs dont work :?
+			rz_cons_printf("ahs %d @ 0x%08" PFMT64x "\n", step, offset + i);
+		} else {
+			rz_cons_printf("0x%08" PFMT64x " -> 0x%08" PFMT64x "\n", offset + i, addr);
 		}
 	}
-	return ptr;
 }
 
-static void cmd_print_bars(RzCore *core, const char *input) {
-	bool print_bars = false;
-	ut8 *ptr = NULL;
-	// p=e [nblocks] [totalsize] [skip]
-	int nblocks = -1;
-	ut64 totalsize = UT64_MAX;
-	int skipblocks = -1;
-	RzIOMap *map;
-	RzListIter *iter;
-	ut64 from = 0, to = 0;
-	RzList *list = rz_core_get_boundaries_prot(core, -1, NULL, "zoom");
-	if (!list) {
-		goto beach;
+static void pr_bb(RzCore *core, RzAnalysisFunction *fcn, RzAnalysisBlock *b, bool emu, ut64 saved_gp, ut8 *saved_arena, char p_type, bool fromHere) {
+	bool show_flags = rz_config_get_i(core->config, "asm.flags");
+	const char *orig_bb_middle = rz_config_get(core->config, "asm.bb.middle");
```

Comment:
OMG. this works because we are so lucky.
```suggestion
	const bool orig_bb_middle = rz_config_get_b(core->config, "asm.bb.middle");
```

Reply:
To be clear. this pointer should be freed the first time the value is changed if this was a dynamic string type, but since it is a boolean, we are very very lucky that never created issues.